featuretools_tsfresh_primitives/primitives/fft_aggregated.py

```diff
@@ -30,3 +35,115 @@ def function(x):
+        min_periods (int, optional): Minimum number of observations required for performing calculations
+            over the window. Can only be as large as window_length when window_length is an integer.
+            When window_length is an offset alias string, this limitation does not exist, but care should be taken
+            to not choose a min_periods that will always be larger than the number of observations in a window.
+            Defaults to 1.
+
+    Note:
+        Only offset aliases with fixed frequencies can be used when defining gap and window_length.
+        This means that aliases such as `M` or `W` cannot be used, as they can indicate different
+        numbers of days. ('M', because different months are different numbers of days;
+        'W' because week will indicate a certain day of the week, like W-Wed, so that will
+        indicate a different number of days depending on the anchoring date.)
+
+    Note:
+        When using an offset alias to define `gap`, an offset alias must also be used to define `window_length`.
+        This limitation does not exist when using an offset alias to define `window_length`. In fact,
+        if the data has a uniform sampling frequency, it is preferable to use a numeric `gap` as it is more
+        efficient.
+
+    Docstring source:
+    https://tsfresh.readthedocs.io/en/latest/api/tsfresh.feature_extraction.html#tsfresh.feature_extraction.feature_calculators.fft_aggregated
+
+    Examples:
+        >>> import pandas as pd
+        >>> datetimes = pd.date_range(start="2019-01-01", freq="1D", periods=10)
+        >>> stffta = ShortTermFftAggregated(aggtype="centroid")
+        >>> output = stffta(datetimes, [1, 2, 4, 8, 4, 2, 1, 2, 4, 8]).tolist()
+        >>> [round(x, 5) for x in output]
+        [0.0, 0.25, 0.27429, 0.27429, 0.2, 0.27429, 0.27429, 0.16667, 0.27429, 0.27429]
+
+        We can also control the gap before the rolling calculation.
+
+        >>> stffta = ShortTermFftAggregated(aggtype="centroid", gap=1)
+        >>> output = stffta(datetimes, [1, 2, 4, 8, 4, 2, 1, 2, 4, 8]).tolist()
+        >>> [round(x, 5) for x in output]
+        [nan, nan, nan, 0.27429, 0.27429, 0.2, 0.27429, 0.27429, 0.16667, 0.27429]
+
+        We can also control the aggregation type.
+
+        >>> stffta = ShortTermFftAggregated(aggtype="variance")
+        >>> output = stffta(datetimes, [1, 2, 4, 8, 4, 2, 1, 2, 4, 8]).tolist()
+        >>> [round(x, 5) for x in output]
+        [0.0, 0.1875, 0.19906, 0.19906, 0.16, 0.19906, 0.19906, 0.13889, 0.19906, 0.19906]
+    """
+
+    name = "short_term_fft_aggregated"
+    input_types = [
+        ColumnSchema(logical_type=Datetime, semantic_tags={"time_index"}),
+        ColumnSchema(semantic_tags={"numeric"}),
+    ]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={"numeric"})
+
+    def __init__(self, aggtype, window_length=3, gap=0, min_periods=0):
```

Comment:
Do you think we should provide a default for `aggtype`? I think we generally (always?) provide default values for primitive arguments.

Reply:
hmm I actually didn't notice this until now. I'll update these primitives as well as the orginal ones